microsoft/testsuites/xfstests/xfstests.py

```diff
@@ -374,56 +548,106 @@ def create_send_subtest_msg(
+                info["information"]["test_section"] = test_section
+            if data_disk:
+                info["information"]["data_disk"] = data_disk
+            info["information"]["test_details"] = str(
+                self.create_xfstest_stack_info(
+                    result.name, test_section, str(result.status.name)
+                )
+            )
             send_sub_test_result_message(
                 test_result=test_result,
                 test_case_name=result.name,
                 test_status=result.status,
+                test_message=result.message,
                 other_fields=info,
             )
 
     def check_test_results(
         self,
         log_path: Path,
-        test_type: str,
-        result: TestResult,
+        test_section: str,
+        result: "TestResult",
         data_disk: str = "",
     ) -> None:
+        """
+        About: This method is intended to be called by run_test method only.
+        This method will check the xfstests output and send subtest results
+        to the test result object.
+        This method depends on create_send_subtest_msg method to send
+        subtest results.
+        Parameters:
+        log_path: The path where the xfstests logs will be saved
+        test_section: The test group name used for testing
+        result: The test result object to which the subtest results will be sent
+        data_disk: The data disk used for testing ( Method partially implemented )
+        """
         xfstests_path = self.get_xfstests_path()
         console_log_results_path = xfstests_path / "xfstest.log"
         results_path = xfstests_path / "results/check.log"
         fail_cases_list: List[str] = []
-
         try:
             if not self.node.shell.exists(console_log_results_path):
                 self._log.error(
                     f"Console log path {console_log_results_path} doesn't exist, please"
                     " check testing runs well or not."
                 )
+                raise LisaException(
+                    f"Console log path {console_log_results_path} doesn't exist, "
+                    "please check testing runs well or not."
+                )
             else:
                 log_result = self.node.tools[Cat].run(
                     str(console_log_results_path), force_run=True, sudo=True
                 )
                 log_result.assert_exit_code()
                 ansi_escape = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
                 raw_message = ansi_escape.sub("", log_result.stdout)
-                self.create_send_subtest_msg(result, raw_message, test_type, data_disk)
+                self.create_send_subtest_msg(
+                    test_result=result,
+                    raw_message=raw_message,
+                    test_section=test_section,
+                    data_disk=data_disk,
+                )
 
             if not self.node.shell.exists(results_path):
                 self._log.error(
```

Comment:
Remove the error line, if it needs to raise an exeception. The error line is not needed.

Reply:
Another place needs to be updated too. Please check all related places for each code comment.